UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -0,0 +1,627 @@
+          PciRootBridgeInfo->RootBridge[index].UID                  = index;
+          PciRootBridgeInfo->RootBridge[index].HID                  = EISA_PNP_ID (0x0A03);
+
+          Data32                                                = (UINT32 *)(PropertyPtr->Data);
+          PciRootBridgeInfo->RootBridge[index].Mem.Base         = Fdt32ToCpu (*(Data32 + 2));
+          PciRootBridgeInfo->RootBridge[index].Mem.Limit        = Fdt32ToCpu (*(Data32 + 6)) + Fdt32ToCpu (*(Data32 + 2)) -1;
+          PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base  = Fdt32ToCpu (*(Data32 + 9)) + LShiftU64 (Fdt32ToCpu (*(Data32 + 8)), 32);
+          PciRootBridgeInfo->RootBridge[index].MemAbove4G.Limit = PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base + LShiftU64 (Fdt32ToCpu (*(Data32 + 12)), 32) +  Fdt32ToCpu (*(Data32 + 13)) -1;
+          PciRootBridgeInfo->RootBridge[index].Io.Base          = Fdt32ToCpu (*(Data32 + 16));
+          PciRootBridgeInfo->RootBridge[index].Io.Limit         = PciRootBridgeInfo->RootBridge[index].Io.Base + Fdt32ToCpu (*(Data32 + 20)) -1;
+
+          DEBUG ((DEBUG_INFO, "RootBridgeCount %x, index :%x\n", RootBridgeCount, index));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Mem.Base %x, \n", PciRootBridgeInfo->RootBridge[index].Mem.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Mem.limit %x, \n", PciRootBridgeInfo->RootBridge[index].Mem.Limit));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->MemAbove4G.Base %llx, \n", PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->MemAbove4G.limit %llx, \n", PciRootBridgeInfo->RootBridge[index].MemAbove4G.Limit));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Io.Base %llx, \n", PciRootBridgeInfo->RootBridge[index].Io.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Io.limit %llx, \n", PciRootBridgeInfo->RootBridge[index].Io.Limit));
+          if (index > 0) {
+            index--;
+          }
+        }
+
+        if (AsciiStrCmp (TempStr, "bus-range") == 0) {
+          UINT16  *Data16;
+
+          DEBUG ((DEBUG_INFO, "  Found bus-range Property TempLen (%08X)\n", TempLen));
+
+          if (RootBridgeCount == 1) {
+            index = 0;
+          }
+
+          Data16                                               = (UINT16 *)(PropertyPtr->Data);
+          PciRootBridgeInfo->RootBridge[index].Bus.Base        = Fdt16ToCpu (*Data16) & 0xFF;
+          PciRootBridgeInfo->RootBridge[index].Bus.Limit       = Fdt16ToCpu (*(Data16 + 1)) & 0xFF;
+          PciRootBridgeInfo->RootBridge[index].Bus.Translation = 0;
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Bus.Base %x, \n", PciRootBridgeInfo->RootBridge[index].Bus.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Bus.limit %x, \n", PciRootBridgeInfo->RootBridge[index].Bus.Limit));
+        }
+      }
+    } else if (AsciiStrCmp (NodePtr->Name, "DebugPrintErrorLevel") == 0) {

```

Comment:
DebugPrintErrorLevel node is not defined in UPL spec, please remove relevant code here.
If required gEdkiiDebugPrintErrorLevelGuid hob can be passed by CustomFdtNode as it is EDK2 specific implementation.